# Copyright 2023 Intrinsic Innovation LLC

--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -15,7 +15,7 @@
 load("@crates_zenoh_c//:defs.bzl", "all_crate_deps")
 load("@rules_cc//cc:defs.bzl", "cc_library")
 load("@rules_rust//cargo:defs.bzl", "cargo_build_script")
-load("@rules_rust//rust:defs.bzl", "rust_binary", "rust_shared_library")
+load("@rules_rust//rust:defs.bzl", "rust_binary", "rust_static_library")
 load("//cbindgen:cbindgen.bzl", "rust_cbindgen_library")
 
 # Transform the opaque types from a .txt to a .rs file.
@@ -40,7 +40,7 @@
 
 # Generate the rust library (name must match the [lib] in Cargo.toml)
 
-rust_shared_library(
+rust_static_library(
     name = "zenohc",
     srcs = glob(
         ["src/**/*.rs"],
@@ -176,6 +176,6 @@
         ],
         "//conditions:default": [],
     }),
-    local_defines = ["ZENOHC_DYN_LIB"],
+
     visibility = ["//visibility:public"],
 )
